torch_xla/experimental/xla_sharding.py

```diff
@@ -396,7 +404,12 @@ def _get_group_assignment(sharding_type: ShardingType,
     ]
     group_assignment = np.arange(len(replication_groups)).reshape(
         tuple(group_tile_shape)).tolist()
-  return group_assignment, replication_groups
+    _get_group_assignment.cache[cache_key] = totuple(
+        [group_assignment, replication_groups])
+  return tolist(_get_group_assignment.cache[cache_key])
```

Comment:
@yeounoh It might be cleaner to modify `_xla_mark_sharding`'s signature to accept `py::tuple` instead of converting back to list here, wdyt?